### ISSUE TITLE
Deprecate ap argument order

### ISSUE THREAD

**Issue by nomisRev** [arrow-kt/arrow#1957]
In #1898 ap parameter order was changed. This should be deprecated in 0.10.x

**Comment by 1Jajen1** [comment]
Talked to @nomisRev a bit on how we could handle this:
There is a few ways this can be done:
- add `apPipe` to 0.10 and suggest changing to it (or waiting for 0.11 and changing broken code)
- suggest to use `mapN(this, ff) { (a, f) -> f(a) }` instead and skip adding `apPipe` altogether (would remove it from #1898 as well as I don't think it is a good operator to have)

If we end up using `apPipe` we might want to consider two deprecation cycles: One for `ap => apPipe` and one for `apPipe => ap or mapN(..)`.

**Comment by aballano** [comment]
@nomisRev @1Jajen1 @arrow-kt/maintainers Given the proposed solutions it seems number 2 is easier, no? Are there any blockers to do this in order to continue the release?

**Comment by 1Jajen1** [comment]
@aballano it is easier (just need to remove it from the current pr and make a pr with a single deprecation notice over `ap` I think). It just got implemented because that was what the discussion in the previous pr that attempted this lead to.

@pakoito @raulraja You reviewed the old pr that attempted this first, what do you think, should we still go with `apPipe`?

**Comment by nomisRev** [comment]
No, that makes sense to me!

So we can do a deprecation cycle with a message that the signature is going to be changed, and a refactor option to `mapN(this, ff) { (a, f) -> f(a) }` (if that is possible).

**Comment by 1Jajen1** [comment]
Although implementing `apPipe` is not too difficult in 0.10.* as well, but that leaves two cycles

**Comment by aballano** [comment]
> Although implementing `apPipe` is not too difficult in 0.10.* as well, but that leaves two cycles

Yeah, I don't think we want to go through many more deprecation cycles beyond the current one before Meta.

**Comment by 1Jajen1** [comment]
Ok so let's decide what do to:
I don't think `apPipe` is important at all, so imo we should just remove it from #1898 and add a pr for current master that marks `ap` as deprecated, explains the signature change and proposes `mapN(this, ff) { (a, f) -> f(a) }` in the meantime.

Thoughts?

**Comment by aballano** [comment]
Fine with it, [also Simon agreed](https://github.com/arrow-kt/arrow/issues/1957#issuecomment-578153839) so I'd say let's go for it 👍
